src/reconcile/rec_write.c

```diff
@@ -343,6 +343,35 @@ __rec_write_page_status(WT_SESSION_IMPL *session, WT_RECONCILE *r)
 	page = r->page;
 	mod = page->modify;
 
+	/*
+	 * Track the page's maximum transaction ID (used to decide if
+	 * we can evict a clean page and discard its history).
+	 */
+	if (WT_TXNID_LT(mod->rec_max_txn, r->max_txn))
+		mod->rec_max_txn = r->max_txn;
+	if (mod->rec_max_timestamp < r->max_timestamp)
+		mod->rec_max_timestamp = r->max_timestamp;
+
+	/*
+	 * Track the tree's maximum transaction ID (used to decide if
+	 * it's safe to discard the tree). Reconciliation for eviction
+	 * is multi-threaded, only update the tree's maximum transaction
+	 * ID when doing a checkpoint. That's sufficient, we only care
+	 * about the maximum transaction ID of current updates in the
+	 * tree, and checkpoint visits every dirty page in the tree.
+	 */
+	if (!F_ISSET(r, WT_REC_EVICT)) {
+		if (WT_TXNID_LT(btree->rec_max_txn, mod->rec_max_txn))
+			btree->rec_max_txn = mod->rec_max_txn;
+		if (btree->rec_max_timestamp < mod->rec_max_timestamp)
+			btree->rec_max_timestamp = mod->rec_max_timestamp;
+	}
+
+	if (WT_TXNID_LT(btree->tmp_rec_max_txn, mod->rec_max_txn))
+		btree->tmp_rec_max_txn = mod->rec_max_txn;
+	if (btree->tmp_rec_max_timestamp < mod->rec_max_timestamp)
+		btree->tmp_rec_max_timestamp = mod->rec_max_timestamp;
+
```

Comment:
I doubt this change is safe: it was an attempt to capture (modulo races) the largest txn ID / timestamp ever seen during reconciliation in a tree.  Without something like this, we're relying on checkpoint visiting everything to calculate the maximum, otherwise it isn't safe for sweep to discard a tree.